### PR TITLE
Resolve ENG-3038 point-regarding numeric machine_type error/Added support for display name in kpi endpoint

### PR DESCRIPTION
**Pull request by ankitintg**
When trying to call get_machine_type with a numeric value we should not want to encapsulate it with additional quotes as it’s already in String even if it is in numeric format.

So I have made some changes to resolve in this PR.

### CODE REVIEW

**Comment by ankitintg** [comment]
We have added support for display name when trying to `get_kpis_for_asset()`

File: smsdk/client.py [hunk 1]
```diff
@@ -262,6 +262,24 @@ def get_kpis_for_asset(self, **kwargs):
...
+        # This loop is getting system name of machine_type if provided only display name in request.
+        for machine_type in kwargs["asset_selection"]["machine_type"]:
+            source_type = machine_types_df.loc[
+                machine_types_df["source_type_clean"] == machine_type, "source_type"
+            ]
+            if not source_type.empty:
+                source_type = list(set(source_type.to_list()))
+                machine_types.extend(source_type)
+            else:
+                machine_types.append(machine_type)
```
Comment (mklein0) [inline smsdk/client.py:278]: Could this lookup be more efficient?  It strikes me the use of Pandas Dataframes here does not make the code easier to understand or more efficient.  I would prefer to deal with things are sets, or collections.

```python
# Convert display name / source type lookup table from dataframe
alias_tbl = machine_types_df.loc[
    :, ["source_type", "source_type_clean"]
].set_index("source_type_clean").groupby("source_type_clean")["source_type"].apply(set).to_dict()

# At this point my knowledge of pandas transforms fail.
machine_types = []
for machine_type in kwargs["asset_selection"]["machine_type"]:
    machine_types.extend(alias_tbl.get(machine_type, (machine_type,)))
```

**Comment by ankitintg** [comment]
Closing this PR as a separate PR for point number 4 from this ticket-https://sightmachine.atlassian.net/browse/ENG-3270 is created here now- https://github.com/sightmachine/sightmachine-sdk/pull/46

File: smsdk/client.py (hunk 1 above)
Comment (ankitintg) [reply smsdk/client.py]: I've addressed your recommended changes in separate PR for a subtask here- https://github.com/sightmachine/sightmachine-sdk/pull/46

**Pull request closed without merging**